src/components/CustomLink/CustomLink.tsx

```diff
@@ -0,0 +1,59 @@
+import { PATHS } from '@src/router';
+import styles from './CustomLink.module.scss';
+import { NavLink, Link } from 'react-router';
+import classNames from 'classnames';
+import type { ReactNode } from 'react';
+
+type LinkType = 'nav' | 'regular';
+type PathFunction = (..._: string[] | number[]) => string;
+
+type StaticPathProps = {
+  pathType: 'static';
+  path: keyof typeof PATHS;
+  pathParams?: never;
+};
+
+type DynamicPathProps = {
+  pathType: 'dynamic';
+  path: PathFunction;
+  pathParams: string[] | number[];
+};
+
+type CustomLinkProps = (StaticPathProps | DynamicPathProps) & {
+  type: LinkType;
+  children: ReactNode;
+  className?: string;
+};
+
+export const CustomLink = ({
+  pathType,
+  path,
+  type,
```

Comment:
We could have `nav?: boolean` so that we can write `<CustomLink nav>` or `<CustomLink>`. Unless other types are coming?